src/V3EmitJson.cpp

```diff
@@ -0,0 +1,242 @@
+            if (it == sources.begin()) {
+                *of << "\t\t\t\"";
+            } else {
+                *of << ",\n\t\t\t\"";
+            }
+            *of << *it << "\"";
+        }
+        if (!sources.empty()) {
+            *of << "\n";
+        }
+    }
+
+    static void emitOverallJson() {
+        const std::unique_ptr<std::ofstream> of{
+            V3File::new_ofstream(v3Global.opt.makeDir() + "/" + v3Global.opt.prefix() + ".json")};
+        const string name = v3Global.opt.prefix();
+
+        *of << "{\n";
+
+        *of << "\t\"system\": {\n";
+        *of << "\t\t\"perl\": \"" << V3Options::getenvPERL() << "\",\n";
+        *of << "\t\t\"python3\": \"" << V3Options::getenvPYTHON3() << "\",\n";
+        *of << "\t\t\"verilator_root\": \"" << V3Options::getenvVERILATOR_ROOT() << "\",\n";
+        *of << "\t\t\"verilator_solver\": \"" << V3Options::getenvVERILATOR_SOLVER() << "\"\n\t},\n\n";
+
+        *of << "\t\"compilier\": {\n";
+        *of << "\t\t\"cflags\": [";
+
+        for (size_t i = 0; i < v3Global.opt.cFlags().size(); i++) {
+            if (i != 0) {
+                *of << ", ";
+            }
+            *of << v3Global.opt.cFlags()[i];
+        }
+
+        *of << "],\n";
+        *of << "\t\t\"ldlibs\": [";
+
+        for (size_t i = 0; i < v3Global.opt.ldLibs().size(); i++) {
+            if (i != 0) {
+                *of << ", ";
+            }
+            *of << v3Global.opt.ldLibs()[i];
+        }
+
+        *of << "]\n\t},\n\n";
+
+        *of << "\t\"switches\": {\n";
+        *of << "\t\t\"systemC\": " << (v3Global.opt.systemC() ? "true" : "false") << ",\n";
+        *of << "\t\t\"coverage\": " << (v3Global.opt.coverage() ? "true" : "false") << ",\n";
+        *of << "\t\t\"use_timing\": " << (v3Global.usesTiming() ? "true" : "false") << ",\n";
+        *of << "\t\t\"threads\": " << v3Global.opt.threads() << ",\n";
+        *of << "\t\t\"trace\": \"" << (v3Global.opt.trace() ? (v3Global.opt.traceFormat().vcd() ? "vcd" : "fst") : "off") << "\"\n\t},\n\n";
+
+        std::vector<string> classes_fast;
+        std::vector<string> classes_slow;
+        std::vector<string> support_fast;
+        std::vector<string> support_slow;
+        std::vector<string> global;
+        for (AstNodeFile* nodep = v3Global.rootp()->filesp(); nodep;
+             nodep = VN_AS(nodep->nextp(), NodeFile)) {
+            const AstCFile* const cfilep = VN_CAST(nodep, CFile);
+            if (cfilep && cfilep->source()) {
+                if (cfilep->support()) {
+                    if (cfilep->slow()) {
+                        support_slow.push_back(cfilep->name());
+                    } else {
+                        support_fast.push_back(cfilep->name());
+                    }
+                } else {
+                    if (cfilep->slow()) {
+                        classes_slow.push_back(cfilep->name());
+                    } else {
+                        classes_fast.push_back(cfilep->name());
```

Comment:
Duplicates code in V3EmitCMake - instead move to common function? Maybe put common stuff in V3EmitMkBase.cpp.  Likewise much of code below.

Perhaps make a pre-pull that just refactors to pulls the code out of V3EmitCMake into V3EmitMkBase, and doesn't add any new features.

Reply:
I don't like the idea to merge cmake and json generators.  I'll see if it is possible to rework verilator-config.cmake to use JSON output.

Reply:
Note I was suggesting a common function that e.g. builds vectors, not combining otherwise.

Reply:
Since cmake generator is going to be rebased on top of JSON manigest it is not relevant anymore (V3EmitCMake.[h.cpp] will be removed at final stage).